pkgs/development/libraries/grpc/default.nix

```diff
@@ -54,22 +54,26 @@ stdenv.mkDerivation rec {
   buildInputs = [ openssl protobuf ]
     ++ lib.optionals stdenv.isLinux [ libnsl ];
 
-  cmakeFlags = [
-    "-DgRPC_ZLIB_PROVIDER=package"
-    "-DgRPC_CARES_PROVIDER=package"
-    "-DgRPC_RE2_PROVIDER=package"
-    "-DgRPC_SSL_PROVIDER=package"
-    "-DgRPC_PROTOBUF_PROVIDER=package"
-    "-DgRPC_ABSL_PROVIDER=package"
-    "-DBUILD_SHARED_LIBS=ON"
-  ] ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
-    "-D_gRPC_PROTOBUF_PROTOC_EXECUTABLE=${buildPackages.protobuf}/bin/protoc"
-  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) && lib.versionOlder stdenv.cc.cc.version "11.0") [
-    # Needs to be compiled with -std=c++11 for clang < 11. Interestingly this is
-    # only an issue with the useLLVM stdenv, not the darwin stdenv…
-    # https://github.com/grpc/grpc/issues/26473#issuecomment-860885484
-    "-DCMAKE_CXX_STANDARD=11"
-  ];
+  cmakeFlags =
+    let
+      # Needs to be compiled with -std=c++11 for clang < 11. Interestingly this is
+      # only an issue with the useLLVM stdenv, not the darwin stdenv…
+      # https://github.com/grpc/grpc/issues/26473#issuecomment-860885484
+      useLLVMAndOldCC = (stdenv.hostPlatform.useLLVM or false) && lib.versionOlder stdenv.cc.cc.version "11.0";
+      cxxStandard = if useLLVMAndOldCC then "11" else "17";
```

Comment:
I think we can remove this bit https://github.com/NixOS/nixpkgs/pull/167129#issuecomment-1092835945

Reply:
`grpc` fails to link with undefined reference errors when `CMAKE_CXX_STANDARD` is not set to 17. Since the default is 17 in nixpkgs, and arrow-cpp requires 17, then setting this to 17 seems like the right thing to do except in the case that was originally there setting it to 11.

Are you suggesting we can delete the `useLLVMAndOldCC` as well and just set `CMAKE_CXX_STANDARD` to `17`?

Reply:
The default if `CMAKE_CXX_STANDARD` is not set is 14 for grpc 1.48.1, which explains why it needs to be set at all.

Reply:
Looks like that's also the default for the latest version (1.50.1).